src/java/org/apache/cassandra/index/sai/utils/RangeIntersectionIterator.java

```diff
@@ -223,11 +223,11 @@ protected RangeIterator buildIterator()
             if (statistics.isDisjoint())
             {
                 // release posting lists
-                FileUtils.closeQuietly(ranges);
+                FileUtils.closeQuietly(rangeIterators);
```

Comment:
I see that `public RangeIterator.Builder add(List<RangeIterator> ranges)` references `ranges`. While it's not used now, there is a risk for bugs in the future. In looking at the super class, what do you think about removing the overrides for `rangeCount` and the two `add` methods. The real logic in this class is in this `buildIterator` method.

Reply:
Looks like `RangeUnionIterator.Builder` has the same mistake in the two different `add` methods.

Reply:
Oops, looks like I read the methods too quickly. There isn't anything wrong.